src/FFmpegWriter.cpp

```diff
@@ -1933,10 +1994,13 @@ void FFmpegWriter::process_video_packet(std::shared_ptr<Frame> frame) {
 bool FFmpegWriter::write_video_packet(std::shared_ptr<Frame> frame, AVFrame *frame_final) {
 #if (LIBAVFORMAT_VERSION_MAJOR >= 58)
 	ZmqLogger::Instance()->AppendDebugMethod("FFmpegWriter::write_video_packet", "frame->number", frame->number, "oc->oformat->flags", oc->oformat->flags);
+
+	if (AV_GET_CODEC_TYPE(video_st) == AVMEDIA_TYPE_VIDEO && AV_FIND_DECODER_CODEC_ID(video_st) == AV_CODEC_ID_RAWVIDEO) {
 #else
 	ZmqLogger::Instance()->AppendDebugMethod("FFmpegWriter::write_video_packet", "frame->number", frame->number, "oc->oformat->flags & AVFMT_RAWPICTURE", oc->oformat->flags & AVFMT_RAWPICTURE);
 
 	if (oc->oformat->flags & AVFMT_RAWPICTURE) {
+#endif
```

Comment:
(I know this is merged already, but I had one small question.) I guess it would be: do we really need the _non_-4.0 path here at all?

`AVFMT_RAWPICTURE` was finally removed in 4.0, but it was marked deprecated waaaay back in 3.0, in [this commit](https://git.ffmpeg.org/gitweb/ffmpeg.git/commitdiff/34ed5c2e4d9b7fe5c9b3aae2da5599fabb95c02e):
```
From: Luca Barbato <lu_zero@gentoo.org>
Date: Mon, 12 Oct 2015 14:06:07 +0000 (+0200)
Subject: avformat: Do not use AVFMT_RAWPICTURE
X-Git-Tag: n3.1-dev~97^2~347
X-Git-Url: http://git.ffmpeg.org/gitweb/ffmpeg.git/commitdiff_plain/34ed5c2e4d9b7fe5c9b3aae2da5599fabb95c02e

avformat: Do not use AVFMT_RAWPICTURE

There are no formats supporting it anymore and it is deprecated.
Update the documentation accordingly.
---
```

Clearly, even at that time it was already considered legacy code.

Meanwhile, at least in terms of the codec enum, `AV_CODEC_ID_RAWVIDEO` [has existed in its current form since FFmpeg 1.0](https://git.ffmpeg.org/gitweb/ffmpeg.git/commit/36ef5369ee9b336febc2c270f8718cec4476cb85) — and it only didn't exist before then, because that's when it was renamed from `CODEC_ID_RAWVIDEO` to the `AV_` prefix. But it's easily as old as dirt.

So, it sounds like that code we **have** to use with `libavformat > 58` is actually the code we _should_ be using on **all** versions... or at the very least, all versions since 3.0 (libavformat 57) for sure. (And as far as I'm concerned, we should just stop supporting any version earlier than 3.0 — if not later.)

Is there something to recommend sticking with `AVFMT_RAWVIDEO` in pre-4.0 code? Does it give us any advantage? Or is it just legacy code we could be rid of?